### PR TITLE
gRPC API for Distance Matrix

### PR DESCRIPTION
**Pull request by agourlay**
context: https://github.com/qdrant/qdrant/milestone/11

This PR adds gRPC endpoints to the new distance matrix API

### CODE REVIEW

File: lib/api/build.rs [hunk 1]
```diff
@@ -274,7 +274,13 @@ fn configure_validation(builder: Builder) -> Builder {
             ("QueryBatchPoints.timeout", "custom(function = \"crate::grpc::validate::validate_u64_range_min_1\")"),
             ("FacetCounts.collection_name", "length(min = 1, max = 255)"),
             ("FacetCounts.key", "length(min = 1)"),
-            ("FacetCounts.timeout", "custom(function = \"crate::grpc::validate::validate_u64_range_min_1\")")
+            ("FacetCounts.filter", ""),
```
Comment (agourlay) [inline lib/api/build.rs:277]: added missing validation on `FacetCounts.filter`

File: docs/redoc/master/openapi.json [hunk 2]
```diff
@@ -13011,16 +13007,18 @@
             ]
           },
           "sample": {
-            "description": "How many points to select and search within.",
+            "description": "How many points to select and search within. Default is 10.",
```
Comment (agourlay) [inline docs/redoc/master/openapi.json:13010]: I decided to make `sample` and `limit` optional for the sake of consistency with other endpoints.

File: lib/api/build.rs (hunk 1 above)
Comment (coszio) [reply lib/api/build.rs:277]: Thank you!

File: docs/grpc/docs.md [hunk 3]
```diff
@@ -3807,6 +3813,110 @@ For example, if `oversampling` is 2.4 and `limit` is 100, then 240 vectors will
...
 
+<a name="qdrant-SearchMatrixOffsets"></a>
+
+### SearchMatrixOffsets
+
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| offsets_row | [uint64](#uint64) | repeated | Row coordinates of the CRS matrix |
```
Comment (coszio) [inline docs/grpc/docs.md:3824]: After some investigation I realized that CRS stands for something else. What we are doing with offsets is more similar to the COO representation

Comment (agourlay) [reply docs/grpc/docs.md:3824]: Good observation :+1: 
I will get this fixed.

File: lib/api/src/grpc/proto/points.proto [hunk 4]
```diff
@@ -612,6 +612,35 @@ message FacetHit {
...
+
+message SearchMatrixPair {
+  PointId a = 1; // first id of the pair
+  PointId b = 2; // second id of the pair
+  float score = 3; // score of the pair
+}
+
+message SearchMatrixOffsets {
+  repeated uint64 offsets_row = 1; // Row coordinates of the matrix
+  repeated uint64 offsets_col = 2; // Column coordinates ids of the matrix
```
Comment (coszio) [inline lib/api/src/grpc/proto/points.proto:638]: How about 
```suggestion
  repeated uint64 row_indices = 1; // Row indices of the matrix
  repeated uint64 col_indices = 2; // Column indices of the matrix
```

And same for REST

**Pull request merged**